### PR TITLE
Nds 869 expose an api for list of errors for inline form errors

### PR DESCRIPTION
**Pull request by coostenbrug**
Intent: Review to merge

### CODE REVIEW

File: components/src/Checkbox/CheckboxGroup.js [hunk 1]
```diff
@@ -9,39 +9,44 @@ import { Fieldset } from "../Form";
...
+      {getCheckboxButtons(otherProps)}
+      <InlineValidation mt="x1" errorMessage={errorMessage} errorList={errorList} />
+    </Fieldset>
+  );
+};
 
 BaseCheckboxGroup.propTypes = {
-  error: PropTypes.string,
+  errorMessage: PropTypes.string,
+  errorList: PropTypes.oneOfType([PropTypes.string, PropTypes.arrayOf(PropTypes.string)]),
```
Comment (clemensp) [inline components/src/Checkbox/CheckboxGroup.js]: Why support arrays and strings for `errorList`?

File: components/src/Select/Select.js [hunk 2]
```diff
@@ -152,7 +153,7 @@ const Select = ({
         highlightedIndex
       }) => (
         <div style={{ position: "relative" }}>
-          <SelectBox {...getToggleButtonProps({ disabled, error, isOpen })}>
+          <SelectBox {...getToggleButtonProps({ disabled, error: !!(errorMessage || errorList), isOpen })}>
```
Comment (clemensp) [inline components/src/Select/Select.js]: Looks like `!!(errorMessage || errorList)` appears in a *lot* of places.
Definitely an opportunity for cleanup.

Comment (coostenbrug) [reply components/src/Select/Select.js]: Did some cleanup here

File: components/src/Checkbox/CheckboxGroup.js (hunk 1 above)
Comment (coostenbrug) [reply components/src/Checkbox/CheckboxGroup.js]: Removed the support for the string case, I agree it made more sense to enforce it to always be an array

**Review by clemensp — APPROVED** [review]

**Pull request merged**